utils/appliance/__init__.py

```diff
@@ -2149,3 +2149,11 @@ def get_or_create_current_appliance():
 class CurrentAppliance(object):
     def __get__(self, instance, owner):
         return get_or_create_current_appliance()
+
+
+class Navigatable(object):
+
+    appliance = CurrentAppliance()
```

Comment:
Do we really want to have this assigned on module import?

Edit: Ah right, its a descriptor class...